base/server/upgrade/10.8.2/01-FixECAdminCertProfile.py

```diff
@@ -30,6 +30,24 @@ def upgrade_subsystem(self, instance, subsystem):
         path = subsystem.config.get('profile.caECAdminCert.config')
         logger.info('Current path: %s', path)
 
+        if path is None:
+            # The attribute is missing. Patch the missing attribute
+            logger.info('profile.caECAdminCert.config missing in CS.cfg')
+
+            path = "{0}/profiles/{1}/caAdminCert.cfg".format(
+                subsystem.base_dir, subsystem.name)
+
+            subsystem.config['profile.caECAdminCert.class_id'] = 'caEnrollImpl'
```

Comment:
We need to consider the workaround that we've suggested people to do. If they only added the missing path but not fix the other properties, those properties will not get fixed because this code only runs `if path is None`. So I think we need to move the code that fixes the `class_id` and `profile.list` outside of `if path is None`.